fairing/http_utils.py

```diff
@@ -0,0 +1,28 @@
+import fairing
+import googleapiclient
+import httplib2
+from fairing.constants import constants
+
+def configure_http_instance(http=None):
+       if not http:
+              http = httplib2.Http()
+       
+       request_orig = http.request
+       user_agent = constants.DEFAULT_USER_AGENT.format(VERSION=fairing.__version__)
+       # Reference: https://github.com/googleapis/google-api-python-client/blob/master/googleapiclient/http.py
+       # The closure that will replace 'httplib2.Http.request'.
+       def new_request(*args, **kwargs):
+              """Modify the request headers to add the user-agent."""
+              if args and len(args)>=4:
+                     headers = args[3] or {}
+                     headers['user-agent'] = user_agent
```

Comment:
In case the request already has a User-Agent, should this be overwriting or appending?https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent 
Possibly appending with a space delimiter might be better?

Reply:
i did that first then looking at how attribution is done, it seems to be done based on regex match so I thought it might get duplicate attribution with google python client. On the second thought that might be ok.